extensions/objects/templates/literally-canvas-diagram-editor.component.spec.ts

```diff
@@ -16,19 +16,94 @@
  * @fileoverview Unit tests for the literally canvas diagram editor.
  */
 
+import { AppConstants } from 'app.constants';
+
 describe('LiterallyCanvasDiagramEditor', function() {
+  var alertSpy = null;
+  var ecs = null;
+  var CsrfService = null;
   var LCDiagramEditorCtrl = null;
+  var $scope = null;
+  var defaultsvg = (
+    '<svg xmlns="http://www.w3.org/2000/svg" width="450" height="350" view' +
+    'Box="0 0 450 350"> <rect width="450" height="350" x="0" y="0" fill="t' +
+    'ransparent" /> <g transform="translate(0, 0)">  </g> </svg>');
+  var linesvg = (
+    '<svg xmlns="http://www.w3.org/2000/' +
+    'svg" width="450" height="350" viewBox="0 0 450 350"> <rect width="450" ' +
+    'height="350" x="0" y="0" fill="transparent" /> <g transform="translate(' +
+    '0, 0)"> <g id="line-241c7047-7297-9aa7-486d-818cfebd30d7"><line x1="105' +
+    '.5" y1="97.125" x2="145.5" y2="130.125" stroke="hsla(0, 0%, 0%, 1)" fil' +
+    'l="undefined" stroke-width="2" stroke-linecap="round"></line></g> </g> ' +
+    '</svg>');
+  var dataUrl = 'data:image/svg+xml;utf8,' + linesvg;
+
   var mockLiterallyCanvas = {
+    currentSvg: defaultsvg,
+    _shapesInProgress: [],
     setImageSize: function(width, height) {
       var text = (
         'The updated diagram width is ' + width +
         ' and height is ' + height);
       return text;
+    },
+    getSVGString: function() {
+      return this.currentSvg;
+    },
+    teardown: function() {
+      return 'LiterallyCanvas is removed';
+    }
+  };
+
+  var mockabas = {
```

Comment:
Hard to understand these. Use full names: mockAssetsBackendApiService, etc.

Reply:
Done.